tests/replication/_base.py

```diff
@@ -53,7 +54,7 @@ def prepare(self, reactor, clock, hs):
         server_factory = ReplicationStreamProtocolFactory(hs)
         self.streamer = hs.get_replication_streamer()
         self.server: ServerReplicationStreamProtocol = server_factory.buildProtocol(
-            None
+            IPv4Address("TCP", "127.0.0.1", 0)
```

Comment:
Aside: slightly surprised to see TCP being part of an IP address struct. Isn't TCP a layer up from IP? /shrug

Reply:
Twisted considers addresses to be many different things: IP, pipes, files, etc.